model/Image.php

```diff
@@ -163,8 +163,35 @@ public function getTag() {
 	public function forTemplate() {
 		return $this->getTag();
 	}
+	
+	/**
+	 * Gets the source image URL for this resource
+	 * 
+	 * @return string
+	 */
+	public function getSourceURL() {
+		return parent::getURL();
+	}
 
 	/**
+	 * Gets the relative URL accessible through the web. If forced resampling is enabled 
+	 * the URL will point to an optimised file, if it is smaller than the original
+	 *
+	 * @uses Director::baseURL()
+	 * @return string
+	 */
+	public function getURL() {
+		if ($this->config()->force_resample) {
+			//return $resampled->getURL();
+			$resampled = $this->getFormattedImage('Resampled');
+			if ($resampled->getAbsoluteSize() < $this->getAbsoluteSize()) {
```

Comment:
I know this is a bit stupid, but what if you wanted to resample something from `100px by 100px` up to `700px by 88px` - the resultant file would be larger and also the one you want.

Yes, you should be setting widths in CSS or somewhere to target that but you could also not. If somethings being resampled then it should always be returned as the resampled one I think?